src/GridComponent.svelte

```diff
@@ -119,9 +129,80 @@
       // Update the active cell's rowIndex and cellIndex
       activeRowIndex = newRowIndex;
       activeCellIndex = newCellIndex;
-      setActiveCell(activeRowIndex, activeCellIndex);
+      setActiveCell(activeRowIndex, activeCellIndex, false);
+    } else if (event.key === "Backspace") {
+      // If the current square is blank, move the cursor back.
+      // If it's not already blank, movement is handled by on:input.
+      if (
+        displaySquares[activeRowIndex][activeCellIndex].content != null &&
+        displaySquares[activeRowIndex][activeCellIndex].content != ""
+      ) {
+        return;
+      }
+
+      event.preventDefault(); // Don't let the backspace affect other content.
+      let activeSpan = activeClue?.span_info;
+      if (activeSpan == null) {
```

Comment:
General linting thing, when checking equality like in js it's normally better to use `===` since it does literal comparison whereas `==` can do some unwanted coercion.

https://stackoverflow.com/questions/523643/difference-between-and-in-javascript